context/param/methodparams.go

```diff
@@ -0,0 +1,69 @@
+package param
+
+import (
+	"fmt"
+	"strings"
+)
+
+//MethodParam keeps param information to be auto passed to controller methods
+type MethodParam struct {
+	name     string
+	location paramLocation
```

Comment:
You mean like this?

type MethodParam struct {
	name         string
	**type         paramType**
	required     bool
	defaultValue string
}